bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1258,15 +1251,35 @@ public void setEnabled(ThingUID thingUID, boolean enabled) {
         }
     }
 
+    private void persistThingEnableStatus(ThingUID thingUID, boolean enabled) {
+        if (storage == null) {
+            logger.debug("Cannot persist enable status of thing with UID {}. Persistent storage unavailable.", thingUID);
+            return;
+        }
+
+        logger.debug("Thing with UID {} will be persisted as {}.", thingUID, enabled ? "enabled." : "disabled.");
+        if (enabled) {
+            // Clear the disabled thing storage. Otherwise the handler will NOT be initialized later.
+            storage.remove(thingUID.getAsString());
+        } else {
+            // Mark the thing as disabled in the storage.
+            storage.put(thingUID.getAsString(), enabled);
+        }
+    }
+
     @Override
     public boolean isEnabled(ThingUID thingUID) {
         Thing thing = getThing(thingUID);
+        if (thing != null) {
+            return thing.isEnabled();
+        }
 
-        if (thing == null) {
-            throw new IllegalArgumentException(String.format("Thing with the UID '%s' is unknown, cannot get its enabled status.", thingUID));
+        logger.debug("Thing with UID {} is unknown. Will try to get the enabled status from the persistent storage.");
+        if (storage != null) {
+            return storage.get(thingUID.getAsString());
```

Comment:
`storage.get()` will return null if there is no entry, i.e. when the thing is not disabled. So we should check the result of `storage.get()` here and return true if it is null.